compiler+runtime/src/cpp/jank/runtime/core.cpp

```diff
@@ -600,4 +600,32 @@ namespace jank::runtime
     return o->type == object_type::tagged_literal;
   }
 
+  object_ptr parse_uuid(object_ptr const o)
```

Comment:
`parse-uuid` on the jvm is odd in that it will throw if the input value is the wrong type, but will return nil if the string is invalid. However, using `#uuid "invalid-string"` will throw an exception. Not sure how much of the exception behavior we want to mirror of the jvm, but I tried to stay close.